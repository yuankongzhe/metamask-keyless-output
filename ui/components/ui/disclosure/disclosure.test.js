import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Disclosure from './disclosure';

describe('Disclosure', () => {
  it('matches snapshot without title prop', () => {
    const { container } = render(<Disclosure variant="">Test</Disclosure>);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with title prop', () => {
    const { container } = render(
      <Disclosure title="Test Title" size="small">
        Test
      </Disclosure>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders content', () => {
    const { container, getByText, rerender } = render(
      <Disclosure title="Test Title">Test</Disclosure>,
    );
    expect(getByText('Test Title')).toBeInTheDocument();
    expect(container.querySelector('.disclosure__content').textContent).toBe(
      'Test',
    );

    expect(
      container.querySelector('.disclosure__content.normal'),
    ).toBeInTheDocument();

    rerender(
      <Disclosure title="Test Title" size="small">
        Test
      </Disclosure>,
    );

    expect(
      container.querySelector('.disclosure__content.small'),
    ).toBeInTheDocument();
  });

  describe('when clicking on disclosure', () => {
    it('does not scroll down on open by default or when isScrollToBottomOnOpen is false', () => {
      const mockScrollIntoView = jest.fn();
      const originalScrollIntoView =
        window.HTMLElement.prototype.scrollIntoView;
      window.HTMLElement.prototype.scrollIntoView = mockScrollIntoView;

      const { getByTestId } = render(
        <Disclosure title="Test Title">Test</Disclosure>,
      );
      const element = getByTestId('disclosure');
      fireEvent.click(element);
      expect(mockScrollIntoView).not.toHaveBeenCalled();
      window.HTMLElement.prototype.scrollIntoView = originalScrollIntoView;
    });

    it('scrolls down on open when isScrollToBottomOnOpen is true', () => {
      const mockScrollIntoView = jest.fn();
      const originalScrollIntoView =
        window.HTMLElement.prototype.scrollIntoView;
      window.HTMLElement.prototype.scrollIntoView = mockScrollIntoView;

      const { getByTestId } = render(
        <Disclosure title="Test Title" isScrollToBottomOnOpen>
          Test
        </Disclosure>,
      );
      const element = getByTestId('disclosure');

      fireEvent.click(element);
      expect(mockScrollIntoView).toHaveBeenCalledWith({
        behavior: 'smooth',
      });
      window.HTMLElement.prototype.scrollIntoView = originalScrollIntoView;
    });
  });
});
