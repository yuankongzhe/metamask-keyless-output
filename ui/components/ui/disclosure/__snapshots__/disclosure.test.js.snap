// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclosure matches snapshot with title prop 1`] = `
<div>
  <div
    class="disclosure"
    data-testid="disclosure"
  >
    <details>
      <summary
        class="disclosure__summary"
      >
        <span
          class="mm-box disclosure__summary--icon mm-icon mm-icon--size-sm mm-box--margin-inline-end-2 mm-box--display-inline-block mm-box--color-inherit"
          style="mask-image: url('./images/icons/add.svg');"
        />
        Test Title
      </summary>
      <div
        class="disclosure__content small"
      >
        Test
      </div>
      <div
        class="disclosure__footer"
      />
    </details>
  </div>
</div>
`;

exports[`Disclosure matches snapshot without title prop 1`] = `
<div>
  <div
    class="disclosure"
    data-testid="disclosure"
  >
    Test
  </div>
</div>
`;
